ArjunaJTS/integration/src/main/java/com/arjuna/ats/jbossatx/jts/ORBWrapper.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2015, Red Hat Middleware LLC, and individual contributors
+ * as indicated by the @author tags.
+ * See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ * This copyrighted material is made available to anyone wishing to use,
+ * modify, copy, or redistribute it subject to the terms and conditions
+ * of the GNU Lesser General Public License, v. 2.1.
+ * This program is distributed in the hope that it will be useful, but WITHOUT A
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License,
+ * v.2.1 along with this distribution; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA  02110-1301, USA.
+ *
+ * @author JBoss Inc.
+ */
+package com.arjuna.ats.jbossatx.jts;
+
+import org.omg.CORBA.NVList;
+import org.omg.CORBA.*;
+import org.omg.CORBA.ORBPackage.InvalidName;
+
+import java.util.Properties;
+import java.applet.Applet;
+
+/**
+ * Wrapper for external orbs to avoid shutting them down when narayana shuts down its own orb
+ */
+class ORBWrapper extends org.omg.CORBA.ORB {
+    private final ORB orb;
+
+    public ORBWrapper(ORB orb) {
+        this.orb = orb;
+    }
+    public static synchronized ORB init() {
+        return ORB.init();
+    }
+    public static ORB init(String[] args, Properties props) {
+        return ORB.init(args, props);
+    }
+    public static ORB init(Applet app, Properties props) {
+        return ORB.init(app, props);
+    }
+    protected void set_parameters(String[] args, Properties props) {
+        ;
+    }
+    protected void set_parameters(Applet app, Properties props) {
+        ;
+    }
+    public void connect(org.omg.CORBA.Object obj) {
+        orb.connect(obj);
+    }
+    public void destroy( ) {
+//        orb.destroy();
```

Comment:
Who is calling destroy at the wrong point then, is it us? Why is destroy ever called at the wrong point?